origin-growth/src/apollo/resolvers.js

```diff
@@ -10,15 +12,63 @@ const resolvers = {
    */
   //JSON: GraphQLJSON,
   DateTime: GraphQLDateTime,
+  GrowthBaseAction: {
+    __resolveType(obj) {
+      if (obj.type === 'referral'){
+        return 'ReferralAction'
+      } else {
+        return 'GrowthAction'
+      }
+    },
+  },
   Query: {
-    async campaigns() {
-      // query campaigns from DB
+    async campaigns(_, args) {
+
+      const campaigns = await Fetcher.getAllCampaigns()
       return {
-        nodes: await db.getCampaigns()
+        totalCount: campaigns.length,
+        nodes: campaigns.map(async (campaign) => await campaign.toApolloObject(args.walletAddress)),
+        pageInfo: {
+          endCursor: 'TODO implement',
+          hasNextPage: false,
+          hasPreviousPage: false,
+          startCursor: 'TODO implement'
+        }
       }
     },
     async campaign() {
       return null
+    },
+    async isEligible(obj, args, context) {
+      if (process.env.NODE_ENV !== 'production') {
+        return {
+          code: 200,
+          success: true,
+          eligibility: 'Eligible',
+          countryName: 'N/A',
+          countryCode: 'N/A'
+        }
+      }
+
+      const locationInfo = await getLocationInfo(context.userIp)
```

Comment:
Note that in the back-end we could simply retrieve the location info directly from the headers that google cloud automatically adds to any request (see this [doc](https://cloud.google.com/appengine/docs/standard/go/reference/request-response-headers) and look for header x-appengine-country .

It would be faster and we would not have to worry about requests quota / paying for the service.. Though downside is that it could be slightly different compared to any location info done by the front-end (unless if switching front-end to use Google Cloud Function geolocation that I had exposed as an experiment).

Reply:
Super helpful! This even removes the need of obtaining the IP address. The only downside is that we can only test this once it is deployed. I guess this will be a thing to test on staging. I will add location data into /#/app-info so we can check there what country has been detected